### PR TITLE
Only add the `ProfileVM.DeleteProfile` handler once

### LINKED ISSUE #13017: Deleting a (last?) profile in the SUI sends the Terminal to a farm upstate
1. Make a new blank profile
2. Save the settings
3. Head over to	that "delete profile" button and hit it
4. [**ZOOP**](https://www.oocities.org/~robink/zoop.jpg), no Terminal

Stack was useless - something above a `DispatcherLoop` with a `ReportUnhandledError` about a "The application attempted to access data outside the valid range" but I apparently don't have `!xamlstowed` installed anymore.

### PR DESCRIPTION
**Pull request by zadjii-msft**
Turns out if you add that Delete handler there, then every time you navigate to the profile, we'll add another Delete handler to the list of handlers. That's bad - that'll cause us to try and delete the profile multiple times.

The repro I had before was 100%, now it's fixed.

* [x] Closes #13017

### CODE REVIEW

File: .github/actions/spelling/allow/allow.txt [hunk 1]
```diff
@@ -38,6 +38,7 @@ hostnames
 hyperlink
 hyperlinking
 hyperlinks
+iconify
```
Comment (carlos-zamora) [inline .github/actions/spelling/allow/allow.txt:41]: THANK YOU!

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 2]
```diff
@@ -419,9 +419,6 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         _PreNavigateHelper();
 
-        // Add an event handler for when the user wants to delete a profile.
-        profile.DeleteProfile({ this, &MainPage::_DeleteProfile });
-
```
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:424]: Ok wait. So, setting `DeleteProfile` multiple times would _append_  an event handler? That seems really unintuitive... wtf. Good find!

**Review by carlos-zamora — APPROVED** [review]

**Review by lhecker — APPROVED** [review]

**Review by DHowett — CHANGES_REQUESTED** [review]

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp [hunk 3]
```diff
@@ -566,6 +563,10 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
                 }
             }
         });
+
+        // Add an event handler for when the user wants to delete a profile.
+        profile.DeleteProfile({ this, &MainPage::_DeleteProfile });
```
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:568]: @DHowett fyi moved from `_Navigate` --> `_CreateProfileNavViewItem` haha

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:424]: That's actually just how winrt events work. Every call appends another handler to the list of handlers, until revoked. It's basically the same as 

```csharp
profile.DeleteProfile += new EventHandler(this._DeleteProfile);
```

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:568]: Yea I think this is okay even without the revoker?

**Review by DHowett — APPROVED** [review]

**Pull request merged**